packages/stats/gbstats/models/statistics.py

```diff
@@ -194,9 +199,93 @@ def create_joint_statistic(
     )
 
 
+@dataclass
+class QuantileStatistic(Statistic):
+    n: int  # number of events here
+    nu: float
+    q_hat: float  # sample estimate
+    q_lower: float
+    q_upper: float
+    main_sum: float  # numerator sum
+    main_sum_squares: float
+    denominator_sum: float  # denominator sum
+    denominator_sum_squares: float
+    main_denominator_sum_product: float
+
+    # Don't we need to similarly handle cases where the sample size and quantile range would make us go < 0 or > 1?
```

Comment:
Yes, I think that's what the original intent of `n` in `_has_zero_variance()` would be, but I guess that requires a bit more context than you have now. maybe we can do that in a follow-up PR.

Reply:
should be included now

Reply:
Suppose n = 100, the distribution is continuous, and `Q(nu_high)` = `Q(nu_low)`.   That means there were no sample observations that fell in about 100*(`nu_high` - `nu_low`)% of the distribution, which is at least 4 percent of the distributional mass (it's smallest for extreme &#nu).  The probability of 0 observations falling in this bin that has roughly 4% of the mass is 0.016.  So this is a low event probability.  In fact, it's even lower, because the separate validation check you mention above would exclude inference for extreme quantile levels if n = 100.

Reply:
Sounds reasonable to me. Thanks for the other checks!